src/sota_tools/server_credentials.h

```diff
@@ -0,0 +1,42 @@
+#ifndef SOTA_CLIENT_TOOLS_SERVER_CREDENTIALS_H_
+#define SOTA_CLIENT_TOOLS_SERVER_CREDENTIALS_H_
+
+#include <string>
+
+enum AuthMethod { AUTH_NONE = 0, AUTH_BASIC, OAUTH2, CERT };
+
+class ServerCredentials{
+    public:
+        ServerCredentials(const std::string &creds);
+        bool CanSignOffline() const { return can_sign_offline_; };
+        AuthMethod GetMethod() const {return method_; };
+        std::string GetClientCert() const { return client_cert_; };
+        std::string GetClientKey() const { return client_key_; };
+        std::string GetRootCert() const { return root_cert_; };
+        std::string GetRepoUrl() const { return repo_url_; };
+        std::string GetAuthUser() const { return auth_user_; };
+        std::string GetAuthPassword() const { return auth_password_; };
+        std::string GetAuthServer() const { return auth_server_; };
+        std::string GetOSTreeServer() const { return ostree_server_; };
+        std::string GetClientId() const { return client_id_; };
+        std::string GetClientSecret() const { return client_secret_; };
+
+
+    private:
+        AuthMethod method_;
+        std::string client_cert_;
+        std::string client_key_;
+        std::string root_cert_;
+        std::string repo_url_;
+        std::string auth_user_;
+        std::string auth_password_;
+        std::string auth_server_;
+        std::string ostree_server_;
+        std::string client_id_;
+        std::string client_secret_;
+        bool can_sign_offline_;
```

Comment:
It would be better to calculate this in the CanSignOffline() method, rather than cache it from the ctor.  That way there will never be any problems keeping the can_sign_offline_ member variable correctly updated, for example if someone adds some new code paths through the constructor is forgets to set can_sign_offline_ (or adds code after that point that sets variables that can_sign_offline_ depends on.

Reply:
I think it is better to cover that with tests.

Reply:
I have submitted tests that tests  CanSignOffline functionality. But If You assure we really need to move that functionality I will do that. Anyway the tests is useful here.

Reply:
Tests are great. I'm definitely keen on keeping those!  It is really just the redundant information that we are storing that I'm worried about.  If we just calculate `CanSignOffline` every time the answer is needed, then there is no chance that the two can get out of sync.

Reply:
Ok. done